scripts/pyi_generator.py

```diff
@@ -658,6 +660,12 @@ def scan_all(self, targets):
                     for file in files:
                         if file in EXCLUDED_FILES or not file.endswith(".py"):
                             continue
+                        if (
+                            changed_files is not None
+                            and os.path.relpath(os.path.join(root, file), ".")
```

Comment:
overall i'd prefer to see _less_ usage of the older `os` APIs which aren't as nice to work with as `pathlib.Path`.

This can be written as `(Path(root) / file).relative_to(Path(".").resolve())`.

Which similar join already appears on line 669 so that could become a loop-local var and `Path(".").resolve()` isn't going to change at runtime, so that could be done once outside the loop.